logstreams/src/main/java/io/zeebe/logstreams/rocksdb/RocksDBController.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.logstreams.rocksdb;
+
+import io.zeebe.logstreams.impl.Loggers;
+import io.zeebe.util.ByteValue;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import org.agrona.CloseHelper;
+import org.rocksdb.*;
+import org.slf4j.Logger;
+
+/**
+ * Controls opening, closing, and managing of RocksDB associated resources. Could be argued that db
+ * reference should not be made transparent, as it could be closed on its own elsewhere, but for now
+ * it's easier.
+ *
+ * <p>Current suggested method of customizing RocksDB instance per stream processor is to subclass
+ * this class and to override the protected methods to your liking.
+ *
+ * <p>Another option would be to use a Builder class and make RocksDBController entirely controlled
+ * through its properties.
+ */
+public class RocksDBController implements AutoCloseable {
+  private static final Logger LOG = Loggers.ROCKSDB_LOGGER;
+
+  private boolean isOpened = false;
+  private RocksDB db;
+  private File dbDirectory;
+  private List<AutoCloseable> closeables = new ArrayList<>();
+
+  static {
+    RocksDB.loadLibrary();
+  }
+
+  public void delete() throws Exception {
+    delete(dbDirectory);
+  }
+
+  public void delete(final File dbDirectory) throws Exception {
+    if (isOpened && this.dbDirectory == dbDirectory) {
+      close();
+    }
+
+    try (Options options = createOptions()) {
+      RocksDB.destroyDB(dbDirectory.toString(), options);
```

Comment:
Ah, sorry I never replied here. If you try to destroy an opened database it will throw an error, as that database is in-use (as far as it can tell).

For example:

```java
@Test
  public void testMahThing() throws RocksDBException {
    final String path = new File(temporaryFolder.getRoot(), "db").getAbsolutePath();
    try (final Options options = new Options().setCreateIfMissing(true);
        final RocksDB db = RocksDB.open(options, path)) {
      RocksDB.destroyDB(path, options);
    }
  }
```

Will throw:

> org.rocksdb.RocksDBException: While lock file: /tmp/junit714478496762411782/db/LOCK: No locks available